### DISCUSSION TITLE
TLS/mTLS support for gotenberg

### DISCUSSION THREAD

**Discussion by gilzig** [gotenberg/gotenberg#716] [Q&A]
Hi,

Is there a plan to add TLS/mTLS support for gotenberg? Currently, if someone manages to monitor all the traffic going to gotenberg, they will be able to gain access to all the data, which may contain information we would not like to expose...

Thank you very much!

**Comment by gulien** [comment]
Gotenberg should not be exposed to the external world. In an internal network, it is very common to not using HTTPS; if someone unauthorized has access to it, not having HTTPS is the least of your problem.

That being said, you could put Gotenberg behind a reverse proxy (nginx, caddy, etc.) that handles the HTTPS part for you.

**Reply by gilzig** [reply]
I agree to some level, but security is all about layers... It is recommended to secure connectivity even in internal networks (and even in a docker-compose cluster...). Every layer of security added improves the overall cyber security level of a system. Each service (even those not expected to be exposed to the internet) should protect its connection.
I agree it is usually not critical in internal networks... but it is recommended...
From your answer I understand there is no plan to add TLS support to gotenberg. 
Thank you for the information :-)
